File: src/java/reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String[] temp=s.trim().split(" ");
        String ans="";
        for(int i=temp.length-1;i>0;i--){
            if(temp[i]!="") ans=ans+temp[i]+" ";
        }
        ans=ans+temp[0];
        return ans;
    }
}

File: src/cpp/reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        int n=s.size();
        int i=0;
        vector<string>temp;
        string t="";
        for(int i=0;i<n;i++){
            if(s[i]!=' ') t.push_back(s[i]);
            else{
                if(t!="") temp.push_back(t);
                t="";
            }
        }
        if(t!="") temp.push_back(t);
        string ans="";
        for(int i=temp.size()-1;i>=0;i--){
            ans=ans+temp[i]+' ';
        }
        ans.pop_back();
        return ans;
    }
};
